modules/gapi/include/opencv2/gapi/gcommon.hpp

```diff
@@ -291,6 +292,26 @@ namespace detail
     };
 }
 
+class WorkloadType {
```

Comment:
I don't think that `gcommon.hpp` is a right place to put `WorkloadType` in. This file is far more general for such kind of compile arguments, as `graph_dump_path` and `use_threaded_executor`, mentioned here are actually backend-independent.

Might be some additional file can be placed here: https://github.com/opencv/opencv/tree/4.x/modules/gapi/include/opencv2/gapi/infer. Something like workload_type.hpp?